homeassistant/components/etherrain.py

```diff
@@ -0,0 +1,172 @@
+""".
+
+Support for Etherrain/8.
+
+"""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SSL = False
+DEFAULT_TIMEOUT = 10
+DOMAIN = 'etherrain'
+STATE = 1
+WATER_ON = 2
+WATER_OFF = 3
+
+LOGIN_RETRIES = 2
+
+ER = {}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Etherrain component."""
+    global ER
+    ER = {}
+
+    conf = config[DOMAIN]
+    schema = 'http'
+
+    server_origin = '{}://{}'.format(schema, conf[CONF_HOST])
+    username = conf.get(CONF_USERNAME, None)
+    password = conf.get(CONF_PASSWORD, None)
+
+    ER['server_origin'] = server_origin
+    ER['username'] = username
+    ER['password'] = password
+
+    hass.data[DOMAIN] = ER
+
+    return login()
+
+
+# pylint: disable=no-member
+def login():
```

Comment:
Move the communication with the EtherRain API to a Python module, publish it on PyPI, and make it a requirement.